swatch-contracts/src/main/java/com/redhat/swatch/contract/service/ContractService.java

```diff
@@ -454,20 +454,21 @@ private void deleteMisalignedSubscriptionMeasurements(
   private void deleteSubscriptionMeasurement(
       SubscriptionEntity subscriptionEntity,
       SubscriptionMeasurementEntity subscriptionMeasurementEntity) {
-    log.info(
-        "Deleting subscription_measurement: {} on subscription: {}",
-        subscriptionMeasurementEntity,
-        subscriptionEntity);
+    log.info("Deleting subscription_measurement: {}", subscriptionMeasurementEntity);
     subscriptionMeasurementRepository.delete(subscriptionMeasurementEntity);
     subscriptionEntity.removeMeasurement(subscriptionMeasurementEntity);
     subscriptionMeasurementRepository.flush();
   }
 
   private boolean subscriptionMeasurementsEqual(
       SubscriptionEntity existing, SubscriptionEntity updated) {
-    return Arrays.equals(
-        updated.getSubscriptionMeasurements().toArray(),
-        existing.getSubscriptionMeasurements().toArray());
+    if (existing.getSubscriptionMeasurements().size()
+        != updated.getSubscriptionMeasurements().size()) {
+      return true;
```

Comment:
This should probably be returning false of the existing size != the updated size

Reply:
thanks, fixed and added a test case for it.